### PR TITLE
fix: path should be dropped for append

### PR DESCRIPTION
**Pull request by nicor88**
### Description
This fix https://github.com/dbt-athena/dbt-athena/issues/98
When working with incremental and append model in parquet, my path was not pruned....looking at the code the drop is excluding append mode.
We should drop the s3 path as well when working with append mode as tables are accumulated, used for inserts and drop at the end.

## Models used to test - Optional
<pre>
{{ config(
    materialized='incremental',
    incremental_strategy='append',
    format='parquet',
    s3_data_naming='table'
) }}

select
    user_id,
    status,
    user_name
from {{ ref('parquet_example_1') }}

</pre>

## Checklist
- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [ ] You added unit testing when necessary
- [ ] You added functional testing when necessary

### CODE REVIEW

File: dbt/include/athena/macros/adapters/relation.sql [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {% macro drop_relation(relation) -%}
-  {% if config.get('table_type') != 'iceberg' and config.get('incremental_strategy') != 'append' %}
```
Comment (nicor88) [inline dbt/include/athena/macros/adapters/relation.sql:2]: @Jrmyy looking at gitblame seems that this was added by you, any specific reasons why??

Comment (Jrmyy) [reply dbt/include/athena/macros/adapters/relation.sql:2]: I did not add this, it was here before : https://github.com/dbt-athena/dbt-athena/pull/79/files#diff-72d05504f9995a78fcddb65052d2c04565d0708899e6fc4ab83cbc8cbf7695d0L2

Comment (nicor88) [reply dbt/include/athena/macros/adapters/relation.sql:2]: Mmm I checked in a rush then,sorry about that.
Any concern about removing it?

**Review by Jrmyy — APPROVED** [review]
If your tests are conclusive, feel free to merge, LGTM ✅

**Comment by nicor88** [comment]
The following models were tested:
* parquet table
* parquet incremental append
* parquet incremental insert_overwrite
* iceberg table
* iceberg incremental merge
* iceberg incremental append

All works good!
Also the main branch works, but it leave the s3 path not clean, and thanks to https://github.com/dbt-athena/dbt-athena/pull/88 path is clean before execution, the small change propose by this pr should be better, as leave the S3 bucket cleaner. Of course if there are failures the path will be cleaned, so again PR 88 solve most of the issues, this is a just a cherry on the top.
 also thanks to this PR  we don't have a failure. but the changes proposed

**Pull request merged**